subworkflows/metadata_and_phylotree.nf

```diff
@@ -5,6 +5,71 @@
 include { METADATA_TABLE } from '../modules/metadata_table'
 include { PHYLO_TREE } from '../modules/phylo_tree'
 
+process PREPARE_LOCATION_INPUT {
+    
+    container 'quay.io/microbiome-informatics/genomes-pipeline.python3base:v1.1'
+    
+    input:
+    file gunc_failed_txt
+    path name_mapping_tsv
+    
+    output:
+    path "accession_list.txt", emit: locations_input_tsv
+    
+    script:
+    """
+    prepare_locations_input.py \
+    --gunc-failed ${gunc_failed_txt} \
+    --name-mapping ${name_mapping_tsv} \
+    --output accession_list.txt
+    """
+}
+
+process FETCH_LOCATIONS {
+
+    container 'quay.io/microbiome-informatics/genomes-pipeline.python3base:v1.1'
+    
+    errorStrategy = { task.attempt <= 3 ? 'retry' : 'finish' }
+    
+    input:
+    path accessions_file
+    path geo_metadata
+    
+    output:
+    path '*.locations', emit: locations_tsv
+    path 'warnings.txt', emit: warnings_txt
+    
+    script:
+    """
+    get_locations.py \
+    -i ${accessions_file} \
+    --geo ${geo_metadata}
+    """
+}
+
+process PUBLISH_WARNINGS {
+
+    publishDir(
+    "${params.outdir}/additional_data/intermediate_files/",
+    pattern: "ena_location_warnings.txt",
+    mode: "copy",
```

Comment:
do you need copy here? I guess you need only `publishDir` to report that file
you can also skip that process at all and use saveas in the process that produces `warning_file`
example https://github.com/EBI-Metagenomics/genomes-generation/blob/a475db69fd3649813f80adfc997d04eea474d81e/config/modules.config#L20

Reply:
No need for a process just for this:
```nextflow
FETCH_LOCATIONS.out.warnings_txt.collectFile(
            name: "all_warnings.txt", storeDir: "${params.outdir}/additional_data/intermediate_files/"
        )
```

Reply:
Amazing, thank you! This works and it's a much better solution.